opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -255,11 +257,45 @@ public Boolean checkIfExists(Table table, String baseEntityId, SQLiteDatabase sq
         return false;
     }
 
+    public void populateFormSubmissionIds() {
+        Cursor mCursor = null;
+        if (formSubmissionIds == null)
+            formSubmissionIds = new HashSet<>();
+
+        try {
+            String query = "SELECT "
+                    + event_column.formSubmissionId
+                    + " FROM "
+                    + eventTable.name();
+            mCursor = getReadableDatabase().rawQuery(query, new String[]{});
+            if (mCursor != null) {
+                while (mCursor.moveToNext()) {
+                    formSubmissionIds.add(mCursor.getString(0));
+                }
+            }
+        } catch (Exception e) {
+            Timber.e(e);
+        } finally {
+            if (mCursor != null) {
+                mCursor.close();
+            }
+        }
+
+    }
+
+    public Boolean checkIfExistsByFormSubmissionId(String formSubmissionId) {
+        if (formSubmissionIds != null && StringUtils.isNotBlank(formSubmissionId)) {
+            return !formSubmissionIds.add(formSubmissionId);
+        }
+        return false;
+    }
```

Comment:
Not sure of this, can this just be `formSubmissionIds.contains(formSubmissionId)`

Reply:
`.add` returns true if id does not exist ... i also need to insert the ids to the set when this happens

Reply:
Assuming the set has all the ids that already exist, I don't see the need to add to this list here

Reply:
If i remove this i will need to add a separate `.add` after `.contains`  ... `.add` already tells me if it exists in the set